### ISSUE TITLE
pip official package compiled differently with docker package?

### ISSUE THREAD

**Issue by u2takey** [tensorflow/tensorflow#22339]
I found in docker container, tensorflow is installed from `http://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-1.7.0-cp27-none-linux_x86_64.whl`
not official pip source `https://pypi.org/project/tensorflow/`. 

And i am working on filesystem plugin, a .so file, which works fine with official tensorflow pip package, but with tensorflow from `http://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-1.7.0-cp27-none-linux_x86_64.whl`, i got error `_ZN10tensorflow10FileSystem10FilesExistERKSt6vectorISsSaISsEEPS1_INS_6StatusESaIS6_EE`.

Seems tensorflow from official source compiled differently with `http://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.0.0-cp27-none-linux_x86_64.whl` ?

Why and how to solve this? How can i  compile a filesystem plugin compatible to `http://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.0.0-cp27-none-linux_x86_64.whl`?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by u2takey** [comment]
@wt-huang  hi，this issue seems not quite related with specific version. I produced this issue with version 1.6-1.8, to be more clear: my custom filesystem plugin works fine with tensorflow from `pip install tensorflow==1.6.0`  `pip install tensorflow==1.7.0` `pip install tensorflow==1.8.0`, but get error `_ZN10tensorflow10FileSystem10FilesExistERKSt6vectorISsSaISsEEPS1_INS_6StatusESaIS6_EE ` with `pip install http://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-1.6.0-cp27-none-linux_x86_64.whl` `pip install http://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-1.7.0-cp27-none-linux_x86_64.whl` `pip install http://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-1.8.0-cp27-none-linux_x86_64.whl`

and my bazel version for compiling filesystem plugin 👍  
```
Build label: 0.13.1
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Wed May 23 11:17:23 2018 (1527074243)
Build timestamp: 1527074243
Build timestamp as int: 1527074243
```
bazel command:

```
bazel build -c opt --copt="-fPIC" --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=0" //tensorflow/core/platform/cxs:cxs_file_system.so
```

**Comment by wt-huang** [comment]
@u2takey I installed above TensorFlow versions without encountering any error. Make sure you use the latest bazel version. Please fill out the template, in particular the exact command to reproduce.

**Comment by gunan** [comment]
To check how to use those dockerfiles, please see the readme in that folder:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/docker/README.md

The link you are seeing there should not exist. its version string is `0.0.0`

Those dockerfiles **cannot** be used without the `parameterized_docker_build.sh` script.
For the newer and more flexible dockerfiles, please see https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/dockerfiles

**Comment by u2takey** [comment]
@gunan @wt-huang why not take a minute to read through my issue before you give answer in a hurry? My issue is not a install issue! My issue is caused by : package from different source compiled differently.

**Comment by gunan** [comment]
When I read your issue, I see that you link to this package:
`http://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.0.0-cp27-none-linux_x86_64.whl`
This is an invalid path, that just stays as a placeholder. (looks like part of your report was modified 14 minutes ago)

When we are building dockerfiles using parameterized_docker_build.sh, this placeholder is replaced with the exact same binaries we release through pypi, only copied to GCS. There is no difference in the packages we use in dockerfiles, or releases.

**Comment by u2takey** [comment]
@gunan sorry for confusion, i haven's expressed my question clearly. 
GCS package is same as pypi, but seems different from package in docker image?
I checked docker hub image history `tensorflow/tensorflow:1.7.0` with docker history
```
/bin/sh -c pip --no-cache-dir install /tensorflow-1.7.0-cp27-cp27mu-manylinux1_x86_64.whl
``` 
seems tensorflow is install from local wal, and tensorflow size is 16533200 and symbol i want to find missing

```
-rwxr-xr-x  1 root staff 16533200 Mar 29 17:04 libtensorflow_framework.so*

nm libtensorflow_framework.so|grep _ZN10tensorflow10FileSystem10FilesExistERKSt6vectorISsSaISsEEPS1_INS_6StatusESaIS6_EE
```
But if i uninstall tensorflow an reinstall in again
```
pip uninstall -y tensorflow
pip install tensorflow==1.7.0
```

i get different tensorflow library, and symbol i want to find  apprears
````
16337456 Sep 20 02:45 libtensorflow_framework.so*

 nm libtensorflow_framework.so|grep _ZN10tensorflow10FileSystem10FilesExistERKSt6vectorISsSaISsEEPS1_INS_6StatusESaIS6_EE
0000000000553240 T _ZN10tensorflow10FileSystem10FilesExistERKSt6vectorISsSaISsEEPS1_INS_6StatusESaIS6_EE
```

**Comment by gunan** [comment]
@av8ramit 
I thought we simply did `pip install tensorflow` in docker images.
Do you know if there are any differences in the pip packages docker uses, and we upload to pypi?

**Comment by av8ramit** [comment]
I cannot find the issue now, but another used had brought this up. We were creating the docker images for a few versions (until 1.7 I believe) with a pip whl file that was created separately. We cannot download from pypi because the docker images are built prior to releasing on pypi. The adjustment we made to satisfy this user was that we installed the exact same whl file instead that we upload to pypi.

TLDR: There used to be differences, now they are not.
